src/sonic-yang-models/yang-models/sonic-warm-restart.yang

```diff
@@ -0,0 +1,105 @@
+module sonic-warm-restart  {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/Azure/sonic-warm-restart";
+    prefix wrm;
+
+    description "SONIC WARMRESTART";
+
+    revision 2021-05-24 {
+        description "First Revision";
+    }
+
+    typedef module-name {
+
+        type enumeration {
+
+            enum bgp {
+               description
+                 "BGP module";
+            }
+
+            enum teamd {
+               description
+                 "Teamd module";
+            }
+
+            enum swss {
+               description
+                 "Switch State Services module";
+            }
+
+            enum system {
+               description
+                 "Warm Restart on a System level.";
+            }
+
+        }
+
+        description
+            "The modules supported in warm-restart";
+    }
+
+
+    container sonic-warm-restart {
+        container WARM_RESTART {
+
+            list WARM_RESTART_LIST {
+                key "module";
+
+                leaf module {
+                    type module-name;
+                    description
+                       "Name of the module";
+                }
+
+                leaf bgp_eoiu {
+                    type boolean;
+                    must "current()/../module = 'bgp'" { error-message "bgp_eoiu is only supported for module bgp."; }
+                    description
+                       "BGP End-of-Initial Update (EOIU) signal enable/disable";
+                }
+
+                leaf bgp_timer {
+                    type uint16 {
+                        range "1..3599" {
+                            error-message "Timer must be 1..3599";
+                            error-app-tag timer-invalid;
+                        }
+                    }
+                    must "current()/../module = 'bgp'" { error-message "bgp_timer is only supported for module bgp."; }
+                    description
+                       "Timer value for BGP module";
```

Comment:
It is unclear to me how what are *_timer values was  used for bgp/team/neighsyncd? Can you elaborate?

Reply:
Per definitions present in swss-schema.md, please see below the details.
https://github.com/Azure/sonic-swss/blob/master/doc/swss-schema.md

;Stores system warm start configuration
;Status: work in progress

key                 = WARM_RESTART:name ; name is the name of SONiC docker or "system" for global configuration.

neighsyncd_timer    = 1*4DIGIT          ; neighsyncd_timer is the timer used for neighsyncd during the warm restart.
                                        ; Timer is started after we restored the neighborTable to internal data structures.
                                        ; neighborsyncd then starts to read all linux kernel entries and mark the entries in
                                        ; the data structures accordingly. Once the timer is expired, we will do reconciliation
                                        ; and push the delta to appDB
                                        ; Valid value is 1-9999. 0 is invalid.

bgp_timer           = 1*4DIGIT          ; bgp_timer holds the time interval utilized by fpmsyncd during warm-restart episodes.
                                        ; During this interval fpmsyncd will recover all the routing state previously pushed to
                                        ; AppDB, as well as all the new state coming from zebra/bgpd. Upon expiration of this
                                        ; timer, fpmsyncd will execute the reconciliation logic to eliminate all the staled
                                        ; state from AppDB. This timer should match the BGP-GR restart-timer configured within
                                        ; the elected routing-stack.
                                        ; Supported range: 1-3600.

teamsyncd_timer     = 1*4DIGIT          ; teamsyncd_timer holds the time interval utilized by teamsyncd during warm-restart episodes.
                                        ; The timer is started when teamsyncd starts. During the timer interval teamsyncd
                                        ; will preserver all LAG interface changes, but it will not apply them. The changes
                                        ; will only be applied when the timer expired. During the changes application the stale
                                        ; LAG entries will be removed, the new LAG entries will be created.
                                        ; Supported range: 1-9999. 0 is invalid

Reply:
Thanks, the valid range in this document is not consistent with the range set in swss-schema.md.

Reply:
Not sure how we got into this value, will fix it to be aligned with schema definition and Warm-restart Click config commands.

Reply:
We've fixed the timer values as per warm restart click config commands.